pluggy.py

```diff
@@ -477,14 +489,30 @@ def _verify_hook(self, hook, hookimpl):
                 "Plugin %r\nhook %r\nhistoric incompatible to hookwrapper" %
                 (hookimpl.plugin_name, hook.name))
 
-        for arg in hookimpl.argnames:
-            if arg not in hook.argnames:
-                raise PluginValidationError(
-                    "Plugin %r\nhook %r\nargument %r not available\n"
-                    "plugin definition: %s\n"
-                    "available hookargs: %s" %
-                    (hookimpl.plugin_name, hook.name, arg,
-                    _formatdef(hookimpl.function), ", ".join(hook.argnames)))
+        # positional arg checking
+        hian, han = set(hookimpl.argnames), set(hook.argnames)
+        notinspec = hian - han
+        notinimpl = han - hian
+
+        if notinspec:
+            raise PluginValidationError(
+                "Plugin %r for hook %r\nhookimpl definition: %s\n%s"
+                "Positional args {0} are declared in the hookimpl but "
+                "can not be found in the hookspec" %
+                (hookimpl.plugin_name, hook.name,
+                 _formatdef(hookimpl.function), notinspec)
+            )
+
+        # warn about new/upcoming arguments to the spec
+        if notinimpl:
```

Comment:
I still find this weird.
Is it supposed to be a feature that you don't have to match the spec?